controllers/component_controller.go

```diff
@@ -293,8 +301,145 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	}
 
+	if component.Spec.Build.ContainerImage == "" {
+		component.Spec.Build.ContainerImage = r.ImageRepository + ":" + component.Namespace + "-" + component.Name
+	}
+
+	_, err = r.generateBuild(ctx, &component)
```

Comment:
Thanks for moving this into a function 👍 .

Could you also move it to the spots I mentioned in https://github.com/redhat-appstudio/application-service/pull/22/files#r793094792?

That'll allow us to properly set the Success or Error status conditions based on the error return value of the function.

E.g. if `generateBuild` fails right now, the status condition will be either a `Created` with a value of `True` or `Updated` with a value of `True`, when the value should actually be `False`.

Reply:
It's a true `generateName` now

https://github.com/redhat-appstudio/application-service/pull/22/files#diff-9f23e6f71fd9e4b8a6dc2c3cda6f3dc30f20ba7519030cbbf46d5085e0dfcdfaR413